connectors/sources/sharepoint_online.py

```diff
@@ -1048,11 +1048,11 @@ async def get_docs_incrementally(self, sync_cursor, filtering=None):
                 "Unable to start incremental sync. Please perform a full sync to re-enable incremental syncs."
             )
 
-        max_data_age = None
+        max_drive_item_age = None
 
         if filtering is not None and filtering.has_advanced_rules():
             advanced_rules = filtering.get_advanced_rules()
-            max_data_age = advanced_rules["maxDataAge"]
+            max_drive_item_age = advanced_rules["dontSubextractDriveItemsOlderThan"]
```

Comment:
Think this can be extracted to a new method

Reply:
Absolutely, we can address it with subsequent PRs